packages/tailwindcss/src/variants.ts

```diff
@@ -95,12 +101,11 @@ export class Variants {
       name,
       (r) => {
         let body = structuredClone(ast)
+        if (usesAtVariant) substituteAtVariant(body, designSystem)
         substituteAtSlot(body, r.nodes)
         r.nodes = body
       },
-      {
-        compounds: compoundsForSelectors(selectors),
-      },
+      { compounds: compoundsForSelectors(selectors) },
```

Comment:
Custom variants that re-use existing `@variant`s are not compoundable. I had an implementation where the `@variant` substitution was happening ahead of time to get the selectors out but we have some checks in `in-*` variant for example that doesn't allow nesting and `@variant …` is implemented using nesting...

I think this is something we can solve in a separate PR and probably after handling flattening ourselves... can of worms.